### PR TITLE
fixed bug with restarted video

### PR DESCRIPTION
**Pull request by PavelAtCarbyne**

### CODE REVIEW

File: plugins/janus_videoroom.c [hunk 1]
```diff
@@ -5010,20 +5010,27 @@ void janus_videoroom_setup_media(janus_plugin_session *handle) {
...
-                         session->is_gst = TRUE;
-                         GError * error = NULL;
-                         g_thread_try_new ("gst", &janus_gst_gst_thread, session, &error);
-                         if (error != NULL) {
-                         	JANUS_LOG (LOG_ERR, "Got error %d (%s) trying to launch the gstreamer gstr thread...\n", error->code, error->message ? error->message : "??");
+                            /*CARBYNE-RF-end*/
+                            /*CARBYNE-GST*/
+                	    JANUS_LOG (LOG_INFO, "BEFORE PIPELINE  PIPELINE--------------------------------%d\n",session->rtpforwardport);
+		            if(g_atomic_int_get(&participant->room->gstrun)) {
+			       JANUS_LOG (LOG_WARN, "RESTART PIPELINE--------------------------------%d\n",session->rtpforwardport);
```
[the remark this answers is not in the document]
Comment (PavelAtCarbyne) [reply plugins/janus_videoroom.c]: done

File: plugins/janus_videoroom.c [hunk 2]
```diff
@@ -5532,15 +5541,17 @@ static void * janus_gst_gst_thread (void * data) {
               session->gstr = gstr;
             }
             else {
-               JANUS_LOG (LOG_ERR, "Invalid gstreamer pipeline..\n");
+               JANUS_LOG (LOG_ERR, "Invalid gstreamer pipeline..   -----%d:%d\n",
```
[the remark this answers is not in the document]
Comment (PavelAtCarbyne) [reply plugins/janus_videoroom.c]: done

File: plugins/janus_videoroom.c [hunk 3]
```diff
@@ -5519,10 +5526,12 @@ static void * janus_gst_gst_thread (void * data) {
     }
    janus_refcount_increase(&session->ref);
    janus_videoroom_publisher *publisher = janus_videoroom_session_get_publisher(session);
-
-    JANUS_LOG (LOG_INFO, "CARBYNE:::::---------------GST 1  --------------\n");
+    unsigned int rtpforwardport = session->rtpforwardport ;
+    JANUS_LOG (LOG_INFO, "CARBYNE:::::---------------GST 1    --%s:%d:%d\n",
+			publisher->room_id_str, rtpforwardport,session->rtpforwardport);
     if (session->is_gst) {
```
[the remark this answers is not in the document]
Comment (PavelAtCarbyne) [reply plugins/janus_videoroom.c]: done

File: plugins/janus_videoroom.c [hunk 4]
```diff
@@ -5010,20 +5010,27 @@ void janus_videoroom_setup_media(janus_plugin_session *handle) {
...
-                         /*CARBYNE-RF-end*/
-                         /*CARBYNE-GST*/
-                         session->is_gst = TRUE;
-                         GError * error = NULL;
-                         g_thread_try_new ("gst", &janus_gst_gst_thread, session, &error);
-                         if (error != NULL) {
-                         	JANUS_LOG (LOG_ERR, "Got error %d (%s) trying to launch the gstreamer gstr thread...\n", error->code, error->message ? error->message : "??");
+                            /*CARBYNE-RF-end*/
+                            /*CARBYNE-GST*/
+                	    JANUS_LOG (LOG_INFO, "BEFORE PIPELINE  PIPELINE--------------------------------%d\n",session->rtpforwardport);
```
[the remark this answers is not in the document]
Comment (PavelAtCarbyne) [reply plugins/janus_videoroom.c]: done

**Review by shmuel7 — APPROVED** [review]

**Pull request merged**